Pynetbox_Data_Uploader/lib/netbox_api/netbox_check.py

```diff
@@ -1,17 +1,13 @@
 from typing import Optional
-from netbox_api.netbox_connect import NetboxConnect
 
 
-class NetboxExistence:
+class NetboxCheck:
     """
     This class contains methods that check if an object exists in Netbox.
     """
 
-    def __init__(self, url: str, token: str, api: Optional = None):
-        if not api:
-            self.netbox = NetboxConnect(url, token).api_object()
-        else:
-            self.netbox = api
+    def __init__(self, netbox: Optional = None):
```

Comment:
Can we type `NetBox` e.g.:
```
def __init__(self, netbox: Netbox = None)
```

Additionally you've marked it as optional, but if a user passes `None` we don't handle it below?